### PR TITLE
GS-303 - added contributing section to gae studio

### PR DESCRIPTION
**Pull request by Pacane**
I copied things from gwtp and adapted a little bit

### CODE REVIEW

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# Contributing Guidelines
+If you are interested in understanding the internals of GAE Studio, building from source, or contributing ideas or modifications to the project, then this document is for you.
+
+#Licensing
```
Comment (meriouma) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: Change all the levels. There should be only one h1 per page

Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: +1

File: site/src/main/markdown/gaestudio/contributing/index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+How to help make this project better...
+
+##Reference
+* [Contributing Guidelines][cg]
+
+#Cloning
+* run `git clone git@github.com:ArcBees/gae-studio.git`
+
+#IDE Setup
+We are using maven to setup the project for both eclipse and IntelliJ.
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/index.md]: `Maven`

Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/index.md]: we should remove Eclipse from our docs, since we don't use it and we don't want external contributors to think we can help them with Eclipse :P

File: site/src/main/markdown/gaestudio/contributing/index.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+##IntelliJ
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/index.md]: I would remove those IntellIj and Eclipse sections.

File: site/src/main/markdown/gaestudio/contributing/index.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+##Build Server
+* [TeamCity Build Server](http://teamcity-private.arcbees.com/project.html?tab=projectOverview&projectId=GaeStudio/) - Snapshots and releases building here
+
+###Directions
+* [Maven Repo Usage Guide](https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide)
+* [Closing a Release Sonatype Directions](https://docs.sonatype.org/display/Repository/Sonatype+OSS+Maven+Repository+Usage+Guide#SonatypeOSSMavenRepositoryUsageGuide-8a.ReleaseIt)
+
+###Notes
+* Important to close eclipse before releasing
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/index.md]: This should be removed

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+  # Decide which code you want to submit. A submission should be a set of changes that addresses one issue in the GAE Studio issue tracker. Please don't mix more than one logical change per submission, because it makes the history hard to follow. If you want to make a change that doesn't correspond to any issue currently listed in the issue tracker, please create one.
+
+Also, coordinate your submission with team members that are listed on the issue in question. This ensures that work isn't being duplicated, and communicating your plan early also generally leads to better patches.
+* Ensure that your code adheres to the GAE Studio source code style.
+* Ensure that there are unit tests for your code.
+* Remember to hg add any new files, and then upload the patch (using [http://codereview.appspot.com/static/upload.py upload.py]) to the Rietveld instance at http://codereview.appspot.com/, Include the relevant issue tracker number in your Rietveld description.
+
+Add your new http://codereview.appspot.com/ url to the issue tracker entry.
+
+###GAE Studio Committers
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: I'd remove this, since committers can easily be displayed using GitHub's stats page.

**Comment by meriouma** [comment]
+1 for removing Eclipse. I think GWTP was setup with eclipse at some point, but not the other projects (and we might have removed eclipse from GWTP too)

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+Unit tests are very important, and we strongly encourage submissions that include them, adding new unit tests for new functionality or updating existing unit tests for bug fixes.
+
+Tests for Java classes should be placed in a parallel source tree under test and the test class name should be suffixed with Test. For example,
+`src/com/google/gwt/core/client/EntryPoint.java` is tested in
+`test/com/google/gwt/core/client/EntryPointTest.java`.
+The use of the parallel test tree has two major advantages:
+  * You can do package scope testing (vs. a tests subpackage).
+  * There is a clear separation between production and test code making packaging easier. This reduces the number of build breaks and the amount of time spent updating build files.
+
+###Submitting Patches
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: this whole `Submitting Patches` should be reviewed:

* Using GitHub, we don't submit patches, we open pull requests
* we don't use codereview.appspot.com anymore
* we don't use `hg` anymore

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+###Parameterized type names
+Parameterized type names should be one capital letter. However, if readability demands longer names (particularly due to having multiple parameters), the name should be capitalized and have suffix "T". In a nutshell, prefer `<T>` or `<K, V>`, and devolve to `<KeyT, ValueT>` if necessary.
+
+|Good|Bad|Tolerable|
+|---|---|
+|`Foo<T>`|`Foo<FooClientType>`| |
+|`Bar<K, V>`|`Bar<KeyType, ValueType>`| |
+|`Baz<V, E, X>`|`Baz<EventType, ErrorType, ExpressionType>`|`Baz<EventT, ErrorT, ExpressionT>`|
+
+###Unit Testing
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: I'd only keep the 1st paragraph.
Explaining the advantages of a parallel source tree should not be documented here, as it's the "standard Maven way of doing things" ™

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+In general, the GAE Studio style is based on the [http://java.sun.com/docs/codeconv/html/CodeConvTOC.doc.html standard Java conventions], except for the differences spelled out below.
+
+For the record, we know that coding style is often a controversial topic. We acknowledge that plenty of great approaches exist out there. We're simply trying to pick one that is at least somewhat consistent with Sun's Java coding conventions, to codify it well, and stick to it.
+
+###Comments and Javadoc
+
+Every file should have an Apache license header at the top, prefaced with a copyright notice. A package statement and import statements should follow, each block separated by a blank line. Next is the class or interface declaration. In the Javadoc comments, describe what the class or interface does.
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: every `.java` file

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md (hunk 1 above)
Comment (Pacane) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: so all # --> ## and ## --> ### ?

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
+# Contributing Guidelines
+If you are interested in understanding the internals of GAE Studio, building from source, or contributing ideas or modifications to the project, then this document is for you.
+
+#Licensing
+All GAE Studio source and pre-built binaries are provided under the Apache 2.0 license.
+
+#The GAE Studio Community
+
+The GAE Studio community exists primarily through mailing lists, the issue tracker and, to a lesser extent, the source control repository. You are definitely encouraged to contribute to the discussion and you can also help us to keep communications effective by following and promoting the guidelines listed here.
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: we don't have a mailing list

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+
+##Checking Out the Source from git
+Run `git clone git@github.com:ArcBees/gae-studio.git`
+
+#Contributing Code
+As GAE Studio is open source, it is easy for our users to fix bugs and create new features, and we always welcome the great patches we get from the community. Before you fire up your favorite IDE and begin hammering away at that new feature, though, please take the time to read this section and understand the process. While it seems rigorous, we want to keep a high standard of quality in the code base.
+
+##Code Style
+To keep the source consistent, readable, diffable and easy to merge, we use a fairly rigid coding style, and all patches will be expected to conform to the style outlined here. To keep things as simple as possible, many of these style rules will be automatically verified by the GAE Studio build; style issues that can't be caught by the build system should be addressed during code review. If you're using Eclipse, we strongly recommend that you install and enable the [http://eclipse-cs.sourceforge.net/ Eclipse Checkstyle plugin] that is used by the provided eclipse configuration.
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: remove the last sentence about Eclipse

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+```
+
+###Class Structure and Member Sort Order
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: This whole `Class Structure and Member Sort Order` section should be revised. 
Ideally, we should only enumerate the differences between the standard Java ordering (as described here http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141855.html#1852) and our coding conventions.

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+    void j() {
+    }
+
+    private void g() {
+    }
+}
+```
+
+###Indentation
+We use 2-space indents for blocks. No tabs at all, anywhere.
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: we use 4 space indents

File: site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+
+```
+Instrument i =
+  new Instrument();
+```
+
+###Imports
+The ordering of import statements is:
+
+* GWT imports
```
Comment (spg) [inline site/src/main/markdown/gaestudio/contributing/Contributing-Guidelines.md]: I don't think this is the right ordering

**Comment by spg** [comment]
LGTM
